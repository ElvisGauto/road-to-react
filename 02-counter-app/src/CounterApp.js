import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value })  => {


    let [ counter, setCounter ] = useState(value);

    // handleAdd
    const handleAdd = (e) => {
        setCounter(counter + 1)
    }

    // handleSubtract
    const handleSubtract = (e) => {
        setCounter(counter - 1)
    }

    // handReset
    const handleReset = (e) => {
        setCounter(value)
    }

    return (
        <>
            <h1>Counter App  Strike</h1>
            <p>{ counter }</p>

            <button onClick={ handleAdd }>+1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ handleSubtract }>-1</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;