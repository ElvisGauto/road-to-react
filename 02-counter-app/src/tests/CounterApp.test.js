import React from 'react';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('<CounterApp /> tests', () => {
    let value;
    let wrapper = shallow(<CounterApp value={value} />);

    beforeEach( () => {
        value = 10;
        wrapper = shallow(<CounterApp value={value} />);
    });

    test('should show <CounterApp /> correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    test('should show default value of props', () => {
        const value = 100;
        const wrapper = shallow(
            <CounterApp 
                value={value}
            />
        );
        const counterText = wrapper.find('p').text();
        expect( Number(counterText) ).toBe( value );
    })

    test('should increase value with button +1', () => {
        wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('p').text();

        expect( counterText ).toBe('11');
    })

    test('should decrease value with button -1', () => {
        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('p').text();

        expect( counterText ).toBe('9');
    })
    
    test('should put a default value with reset button', () => {
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('p').text();

        expect(counterText).toBe('10');
    })
    
    
});