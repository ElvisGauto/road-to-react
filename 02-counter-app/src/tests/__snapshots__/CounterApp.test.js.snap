// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp /> tests should show <CounterApp /> correctly 1`] = `
<Fragment>
  <h1>
    Counter App  Strike
  </h1>
  <p>
    10
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
